iot-commercial-domain/src/main/java/org/dcsa/iot/commercial/domain/persistence/repository/specification/EventCacheSpecification.java

```diff
@@ -32,53 +37,63 @@ public record EventCacheFilters(
     public EventCacheFilters { }
   }
 
-  public static Specification<EventCache> withFilters(final EventCacheFilters filters) {
+  public static Specification<IoTCommercialEvent> withFilters(final EventCacheFilters filters) {
     log.debug("Searching based on {}", filters);
 
-    return (Root<EventCache> root, CriteriaQuery<?> query, CriteriaBuilder builder) -> {
+    return (Root<IoTCommercialEvent> root, CriteriaQuery<?> query, CriteriaBuilder builder) -> {
       List<Predicate> predicates = new ArrayList<>();
-      JsonPathExpressionBuilder jsonPath = new JsonPathExpressionBuilder(root, builder, EventCache_.CONTENT);
-      predicates.add(builder.equal(root.get(EventCache_.EVENT_TYPE), EventType.IOT_COMMERCIAL));
 
       handleParsedQueryParameter(
         predicates,
         builder,
-        root.get(EventCache_.EVENT_CREATED_DATE_TIME),
+        root.get(IoTCommercialEvent_.EVENT_CREATED_DATE_TIME),
         filters.eventCreatedDateTime
       );
 
       handleParsedQueryParameter(
         predicates,
         builder,
-        root.get(EventCache_.EVENT_DATE_TIME),
+        root.get(IoTCommercialEvent_.EVENT_DATE_TIME),
         filters.eventDateTime
       );
 
       if (filters.iotEventTypeCodes != null && !filters.iotEventTypeCodes.isEmpty()) {
-        predicates.add(
-          jsonPath
-            .of(IoTCommercialEvent_.IOT_EVENT_TYPE_CODE)
-            .in(filters.iotEventTypeCodes.stream().map(Enum::name).toList()));
+     //   predicates.add(root.get(IoTCommercialEvent_.IOT_EVENT_TYPE_CODE).in(filters.iotEventTypeCodes.stream().map(Enum::toString).toList()));
```

Comment:
Should either be made to work or removed

Reply:
iotEventTypeCode is added again as filter now.